### DISCUSSION TITLE
docTR and onnxTR don't give same results

### DISCUSSION THREAD

**Discussion by lionelfer** [felixdittrich92/OnnxTR#37] [Q&A] [answered]
Hi,
I try these 2 demo (Felix92/OnnxTR-OCR and mindee/doctr) with same models (db_resnet50+crnn_vgg16_bn) on this image
![capture_1line_url_2](https://github.com/user-attachments/assets/3d28dd43-400c-493d-bef5-010dbd1330e2)
I get different results.  
docTR perform well, it read: url=http://localhost:8000/ocr/"
onnxTR read:  url-http//Localhost:8000/ocr/-
So onnxTR see "-" twice, never read the ", read a "L" instead of "l"

I read the code and didn't point out the pre treatment or the parameters with can explain differences.
I tried a local instance of onnxTR and got the same result as the demo Felix92/OnnxTR-OCR . 
I expect to get same results (with same models !) with Felix92/OnnxTR-OCR  and mindee/doctr.
Can you help me ? am I missing something ?

**Comment by felixdittrich92** [comment]
Hi @lionelfer :wave:,

Unfortunately the Huggingface doctr demo lacks a bit behind (We have already a ticket in docTR to automate the process that the demo is more up to date  https://github.com/mindee/doctr/issues/1090 CC @odulcy-mindee... i tested your image with the given models in docTR and OnnxTR and the results are extactly the same:

```
Det model: db_resnet50
Mean difference between DOCTR and ONNXTR prob maps: 0.00000115
```

```
Render results:

OnnxTR:
url
http://localhost: 8000/ocr/"

docTR: 
url
http://localhost: 8000/ocr/"
```

Btw.: Passing such small already cropped images in tricky (maybe works better with the `recognition_predictor` standalone), because the detection models are trained on documents and every input is resized to 1024x1024 before

**Reply by felixdittrich92** [reply, marked as answer]
Oh ..wrong model (used `db_mobilenet_v3_large` for the render results) now with `db_resnet50` i can reproduce the result in both (docTR & OnnxTR): 

```
url
http - //Localhost: 8000/ocr/ -
```

**Reply by lionelfer** [reply]
Thanks @felixdittrich92, your answer is helpful. i will try the `recognition_predictor` on this kind of image.

**Reply by felixdittrich92** [reply]
@lionelfer 

![test_cropped_2](https://github.com/user-attachments/assets/09ce8d6f-3fde-4151-b8e2-5de509b6f83c)

```
from onnxtr.io import DocumentFile
from onnxtr.models import recognition_predictor

doc = DocumentFile.from_images("/home/felix/Desktop/test_cropped_2.png")

reco_predictor = recognition_predictor("crnn_vgg16_bn")
res = reco_predictor(doc)

print(res)
```

```
[('"http://localhost:8000/ocr/"', 0.5127920508384705)]
```

Keep in mind we don't recognize whitespaces ! :) Also within the detection on a full image this would be 3 parts: ["url", "=", ""http://localhost:8000/ocr/""]
